docs/developers/piece-reference/piece-branching.mdx

```diff
@@ -0,0 +1,57 @@
+---
+title: 'Piece Branching'
+icon: 'hammer'
+description: 'Make a piece branchable'
+---
+Activepieces allows pieces to be directly branched from depending on the action by adding the expected outputs in the `outputs` array of objects when creating an action.
+
+### Waiting for approval action
+
+We will see the branching feature within the context of the approval process, using the waiting for approval action as an example.
+
+**Example:**
+
+```typescript
+import { createAction } from '@activepieces/pieces-framework';
+import { ExecutionType, PauseType, branchedPieceResponse } from '@activepieces/shared';
+
+export const waitForApprovalLink = createAction({
+  name: 'wait_for_approval',
+  displayName: 'Wait for Approval',
+  description: 'Pauses the flow and wait for the approval from the user',
+  props: {},
+  errorHandlingOptions: {
+    continueOnFailure: {
+      hide: true,
+    },
+    retryOnFailure: {
+      hide: true,
+    },
+  },
+  // Here we define the names of the outputs expected to be branched from the piece
+  outputs: [
```

Comment:
Instead of using a name since you are using it as an id

We can change this to an object where the key is the name of the branch

Reply:
I think the issue would be mainly with adding other properties to the branch in case we would need to add this later, if we would go with an object, I suggest this way:
```ts
{
    branchName: {
       expectedOutput: true,
       ...otherProperties,
    },
} 
```
by this way, we would be able to extend the functionality further in the future. What do you think?